### PR TITLE
scripts: build: gen_handles: use Z_DECL_ALIGN

### PR DESCRIPTION
**Pull request by gmarull**
Instead of hardcoding alignment size for pass 2 device handles, use Z_DECL_ALIGN. This makes sure gen_handles.py is always in sync with the type defined in device.h. The build assert in device.h can be removed as a result, since we do not hardcode handles size anywhere else.

Note: this assertion was introduced in #32127, as far as I can see, we do not hardcode the size in any linker script even though the message says, "fix the linker scripts".

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

### CODE REVIEW

**Review by JordanYates — APPROVED** [review]
The assertion is a holdover from the time when the handles array was not allowed to change sizes after the first linker pass, and python scripts were assuming 2 byte handles.

**Review by dcpleung — APPROVED** [review]

File: include/zephyr/device.h [hunk 1]
```diff
@@ -62,13 +62,6 @@ extern "C" {
  */
 typedef int16_t device_handle_t;
 
-/*
- * The build assert will fail if device_handle_t changes size, which
- * means the alignment directives in the linker scripts and in
- * `gen_handles.py` must be updated.
- */
-BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
```
Comment (carlescufi) [inline include/zephyr/device.h:70]: Why did you remove this? This is about the size of `uint16_t` and not the alignment?

Comment (carlescufi) [reply include/zephyr/device.h:70]: OK, saw this comment now https://github.com/zephyrproject-rtos/zephyr/pull/51162#pullrequestreview-1137291177

**Review by carlescufi — APPROVED** [review]

File: include/zephyr/device.h (hunk 1 above)
Comment (gmarull) [reply include/zephyr/device.h:70]: because we relied on size being 2 (hardcoded), like in the gen_handles.py script.
